pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -416,12 +456,94 @@ func newDeployment(etcdstorage *samplev1alpha1.EtcdStorage) *appsv1.Deployment {
 				Spec: corev1.PodSpec{
 					Containers: []corev1.Container{
 						{
-							Name:  "nginx",
-							Image: "nginx:latest",
+							Name:    "etcdproxy",
+							Image:   "quay.io/coreos/etcd:v3.2.18",
+							Command: []string{"/usr/local/bin/etcd", "grpc-proxy", "start"},
+							Args:    []string{etcdaddr, etcdns, "--listen-addr=0.0.0.0:2379", "--cacert=/etc/certs/ca/ca.pem", "--cert=/etc/certs/client/client.pem", "--key=/etc/certs/client/client-key.pem"},
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "etcd",
+									Protocol:      corev1.ProtocolTCP,
+									ContainerPort: 2379,
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      c.etcdConnectionInfo.EtcdCertSecretName,
+									MountPath: "/etc/certs/client",
+									ReadOnly:  true,
+								},
+								{
+									Name:      c.etcdConnectionInfo.EtcdCAConfigMapName,
+									MountPath: "/etc/certs/ca",
+									ReadOnly:  true,
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: c.etcdConnectionInfo.EtcdCertSecretName,
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: c.etcdConnectionInfo.EtcdCertSecretName,
+								},
+							},
+						},
+						{
+							Name: c.etcdConnectionInfo.EtcdCAConfigMapName,
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: c.etcdConnectionInfo.EtcdCAConfigMapName,
+									},
+								},
+							},
 						},
 					},
 				},
 			},
 		},
 	}
 }
+
+func (c *Controller) newService(etcdstorage *samplev1alpha1.EtcdStorage) *corev1.Service {
+	labels := map[string]string{
+		"controller": "epc",
+	}
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("etcd-%s", etcdstorage.Name),
+			Namespace: c.controllerNamespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(etcdstorage, schema.GroupVersionKind{
+					Group:   samplev1alpha1.SchemeGroupVersion.Group,
+					Version: samplev1alpha1.SchemeGroupVersion.Version,
+					Kind:    "EtcdStorage",
+				}),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports: []corev1.ServicePort{
+				{
+					Protocol:   corev1.ProtocolTCP,
+					Port:       2379,
+					TargetPort: intstr.FromInt(2379),
+				},
+			},
+		},
+	}
+}
+
+func GetControllerNamespace() string {
```

Comment:
needs godoc.

Reply:
Added along with some details about the function.